answerking_app/tests/test_unit/test_unit_productserializerreadonly.py

```diff
@@ -0,0 +1,134 @@
+from decimal import Decimal
+
+from rest_framework.utils.serializer_helpers import ReturnDict
+
+from answerking_app.models.models import Category, Product
+from answerking_app.models.serializers import ProductSerializerReadOnly
+from answerking_app.tests.test_unit.UnitTestBaseClass import UnitTestBase
+
+
+class ProductSerializerTests(UnitTestBase):
+    UTB = UnitTestBase()
+    test_cat_data: dict = UTB.get_fixture(
+        "categories", "burgers_cat_data.json"
+    )
+    test_prod_data: dict = UTB.get_fixture(
+        "products", "plain_burger_data.json"
+    )
+
+    def setUp(self):
+        cat: Category = Category.objects.create(**self.test_cat_data)
+        prod: Product = Product.objects.create(**self.test_prod_data)
+        cat.products.add(prod)
+
+    def tearDown(self):
+        Category.objects.all().delete()
+        Product.objects.all().delete()
+
+    def test_product_read_only_serializer_contains_correct_fields(self):
+        test_prod: Product = Product.objects.get(
+            name=self.test_prod_data["name"]
+        )
+        test_serializer_data: ReturnDict = ProductSerializerReadOnly(
+            test_prod
+        ).data
+        expected: list[str] = [
+            "id",
+            "name",
+            "description",
+            "price",
+            "categories",
+        ]
+        actual: list[str] = list(test_serializer_data.keys())
+        self.assertCountEqual(actual, expected)
```

Comment:
don't do countEqual do Equal so we check the fields are the correct ones.

Reply:
Done